### PR TITLE
Edit DG to include new proposed features

### PR DESCRIPTION
**Pull request by GabrielSimbingyang**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -213,9 +213,45 @@ The following activity diagram summarizes what happens when a user executes a ne
...
 
-_{Explain here how the data archiving feature will be implemented}_
+### \[Proposed\] Deck System
+This feature would allow the user to create multiple lists of entries rather than having 
+all translations together in the same list. This means that the user could have different decks
+for different languages or even multiple decks for the same languages. 
+
+
+_{Feature will be added in v1.2}_
+
```
Comment (MelanieNPS) [inline docs/DeveloperGuide.md:228]: The integration between the deck and the UI may take a longer time than we expect, buffer time between thursday and friday may not be enough. Possibly in version 1.3

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -213,9 +213,45 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+### \[Proposed\] Flashcard System
+
+The flashcard system would allow the user to choose to practice in whichever deck
+he wishes.
+
+The methodology behind GreenTea's flashcard system will be based on the Leitner System (https://en.wikipedia.org/wiki/Leitner_system)
+To put it simply, it is a proven quizzing system that increases the user's rate of learning by
```
Comment (MelanieNPS) [inline docs/DeveloperGuide.md]: I think the 'to put it simply' can be removed. Maybe something like " It is a proven quizzing system that increases the user's rate of learning by using spaced repetition. Flashcards are sorted based on the user's ability to answer them.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -213,9 +213,45 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+3. Average time taken in total
+4. Time of quiz
+
+From these data, GreenTea would be able to derive some meaningful analytics to 
+display to the user. These include:
+1. Progression since last attempt
+2. Length of time between quizzes
+3. Most forgotten phrase/translation 
+4. Language mastery 
+
```
Comment (MelanieNPS) [inline docs/DeveloperGuide.md:257]: Looks good!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (GabrielSimbingyang) [reply docs/DeveloperGuide.md]: Noted, will update this

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (GabrielSimbingyang) [reply docs/DeveloperGuide.md:228]: Noted, will update this

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (GabrielSimbingyang) [reply docs/DeveloperGuide.md:257]: 👍

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (GabrielSimbingyang) [reply docs/DeveloperGuide.md]: Updated!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (GabrielSimbingyang) [reply docs/DeveloperGuide.md:228]: Updated!

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -213,9 +213,45 @@ The following activity diagram summarizes what happens when a user executes a ne
 
 _{more aspects and alternatives to be added}_
 
-### \[Proposed\] Data archiving
 
-_{Explain here how the data archiving feature will be implemented}_
+### \[Proposed\] Deck System
+This feature would allow the user to create multiple lists of entries rather than having 
+all translations together in the same list. This means that the user could have different decks
+for different languages or even multiple decks for the same languages.
```
Comment (vigneshbhuvan-nus) [inline docs/DeveloperGuide.md:220]: I think maybe we could provide written examples of what kind of different decks users can have, to help them understand better
For e.g. 
- Spanish
- Japanese

OR

- Spanish Animals
- Spanish Food

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -213,9 +213,45 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+3. Average time taken in total
+4. Time of quiz
+
+From these data, GreenTea would be able to derive some meaningful analytics to 
+display to the user. These include:
+1. Progression since last attempt
+2. Length of time between quizzes
+3. Most forgotten phrase/translation 
+4. Language mastery 
+
```
Comment (vigneshbhuvan-nus) [inline docs/DeveloperGuide.md:257]: Possibly can include 
5. Progress in each Deck (to understand how good a user is in each deck)

**Review by vigneshbhuvan-nus — COMMENTED** [review]
Looks very good, I dropped some comments for you to consider. Lmk what you think. thanks

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (GabrielSimbingyang) [reply docs/DeveloperGuide.md:220]: Done

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (GabrielSimbingyang) [reply docs/DeveloperGuide.md:257]: Done!

**Review by geoboom — COMMENTED** [review]
lgtm

**Pull request merged**